src/Fantomas.Core.Tests/FormattingSelectionOnlyTests.fs

```diff
@@ -1,9 +1,200 @@
 module Fantomas.Core.Tests.FormattingSelectionOnlyTests
 
+open FSharp.Compiler.Text
+open Fantomas.Core
 open NUnit.Framework
 open FsUnit
 open Fantomas.Core.Tests.TestHelper
 
+let private config = FormatConfig.FormatConfig.Default
+
+let private formatSelectionOnly isFsiFile selection (source: string) config =
+    let formattedSelection, _ =
+        CodeFormatter.FormatSelectionAsync(isFsiFile, source, selection, config)
+        |> Async.RunSynchronously
+
+    formattedSelection.Replace("\r\n", "\n")
+
+let private formatSelectionAndAssertRange isFsiFile selection source config =
+    let _, resultRange =
+        CodeFormatter.FormatSelectionAsync(isFsiFile, source, selection, config)
+        |> Async.RunSynchronously
+
+    (resultRange.StartLine, resultRange.StartColumn, resultRange.EndLine, resultRange.EndColumn)
+    == (selection.StartLine, selection.StartColumn, selection.EndLine, selection.EndColumn)
+
+let private mkSelection (startLine, startColumn) (endLine, endColumn) =
+    let startPos = Position.mkPos startLine startColumn
+    let endPos = Position.mkPos endLine endColumn
+    Range.mkRange "selection" startPos endPos
+
+[<Test>]
+let ``SynModuleDecl with exact selection`` () =
+    formatSelectionOnly
+        false
+        (mkSelection (6, 0) (6, 12))
+        """
+module A
+
+let x = 1
+
+let y =    2
+"""
+        config
+    |> should equal "let y = 2"
+
+[<Test>]
+let ``SynExpr with exact selection`` () =
+    formatSelectionOnly
+        false
+        (mkSelection (4, 13) (4, 16))
+        """
+module A
+
+let two =    1+1
+"""
+        config
+    |> should equal "1 + 1"
+
+[<Test>]
+let ``SynExpr with larger selection`` () =
+    formatSelectionOnly
+        false
+        (mkSelection (4, 0) (10, 0))
+        """
+let x =
+    printfn "start"
+
+    try
+        something ()
+    with
+    | ex ->
+        printfn "caught"
+
+    ()
+"""
+        config
+    |> prepend newline
+    |> should
+        equal
+        "
```

Comment:
Because the string ends on a `"` character.
![image](https://user-images.githubusercontent.com/2621499/175807562-f3a5c9c3-1541-4de8-a026-8979f13c7c6f.png)